src/Psalm/Report/ConsoleReport.php

```diff
@@ -32,10 +34,12 @@ private function format(\Psalm\Internal\Analyzer\IssueData $issue_data): string
         }
 
         $issue_reference = $issue_data->link ? ' (see ' . $issue_data->link . ')' : '';
+        $file_basename = basename($issue_data->file_name);
+        $file_path = substr($issue_data->file_name, 0, -strlen($file_basename));
 
         $issue_string .= ': ' . $issue_data->type
-            . ' - ' . $issue_data->file_name . ':' . $issue_data->line_from . ':' . $issue_data->column_from
-            . ' - ' . $issue_data->message . $issue_reference . "\n";
+            . ' - ' . $file_path . "\e[1;31m" . $file_basename . ':' . $issue_data->line_from . "\e[0m"
```

Comment:
It should be conditional on `$this->use_color`. Incidentally this will also fix most of the test failures, as those tests mostly disable colored output.

Reply:
As for testing this specific feature, you could take inspiration from https://github.com/vimeo/psalm/blob/5cf660f63b548ccd4a56f62d916ee4d6028e01a3/tests/ReportOutputTest.php#L1071-L1085